src/main/java/org/jenkinsci/plugins/additionalmetrics/Utils.java

```diff
@@ -99,18 +99,33 @@ static Optional<RunWithDuration> findRun(List<? extends Run> runs, Predicate<Run
     }
 
     static Optional<Duration> averageDuration(List<? extends Run> runs, Predicate<Run> preFilter, ToLongFunction<Run> durationFunction) {
```

Comment:
Hey, let me suggest another way, tell me what you think:

```
    static Optional<Duration> averageDuration(List<? extends Run> runs, Predicate<Run> preFilter, ToLongFunction<Run> durationFunction) {
        return durationFunction(runs, preFilter, durationFunction, LongStream::average);
    }

    static Optional<Duration> stdDevDuration(List<? extends Run> runs, Predicate<Run> preFilter, ToLongFunction<Run> durationFunction) {
        return durationFunction(runs, preFilter, durationFunction, longStream -> /* implement me */);
    }

    static Optional<Duration> durationFunction(List<? extends Run> runs, Predicate<Run> preFilter, ToLongFunction<Run> durationFunction, Function<LongStream, OptionalDouble> durationCollector) {
        LongStream longStream = preFilter(runs, preFilter)
                .filter(r -> durationFunction.applyAsLong(r) > 0)
                .mapToLong(durationFunction);

        OptionalDouble val = durationCollector.apply(longStream);

        if (val.isPresent()) {
            return Optional.of(new Duration((long) val.getAsDouble()));
        } else {
            return Optional.empty();
        }
    }
```

This avoids repeating the whole function, and only passes the final operation (stdev or average, which exists btw, why reimplement it?)